src/interface.rs

```diff
@@ -134,13 +137,17 @@ where
     ///
     /// Most likely there was a mistake with the 2in9 busy connection
     /// //TODO: use the #cfg feature to make this compile the right way for the certain types
-    pub(crate) fn wait_until_idle(&mut self, is_busy_low: bool) {
-        // //tested: worked without the delay for all tested devices
-        // //self.delay_ms(1);
+    pub(crate) fn wait_until_idle(&mut self, delay: &mut DELAY, is_busy_low: bool) {
         while self.is_busy(is_busy_low) {
-            // //tested: REMOVAL of DELAY: it's only waiting for the signal anyway and should continue work asap
-            // //old: shorten the time? it was 100 in the beginning
-            // //self.delay_ms(5);
+            // This has been removed and added many time :
+            // - it is faster to not have it
+            // - it is complicated to pass the delay everywhere all the time
+            // - busy waiting can consume more power that delaying
+            // - delay waiting enables task switching on realtime OS
```

Comment:
exactly what I need!, thanks

Reply:
Be careful, on esp32 you have to use delay::FreeRtos and not delay::Ets and a minimum delay_ms on 10, to make sure there is a task switch

Reply:
ah, thanks, I knew and tried delay::FreeRtos, but didn't know about the 10ms limitation.